### PR TITLE
[#91]feat: implement pgmoneta-cli encrypt

### PR DESCRIPTION
**Pull request by solokyo**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -1225,6 +1225,11 @@ accept_mgt_cb(struct ev_loop* loop, struct ev_io* watcher, int revents)
          pgmoneta_management_write_int32(client_fd, ret);
          free(payload_s1);
          break;
+      case MANAGEMENT_ENCRYPT:
+         pgmoneta_log_debug("pgmoneta: Management encrypt: %s/%s", payload_s1);
```
Comment (jesperpedersen) [inline src/main.c:1229]: There is only one argument - remove `/%s`

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
Could you update `GETTING_STARTED.md` and `pgmoneta.spec` as well ?

File: src/main.c (hunk 1 above)
Comment (solokyo) [reply src/main.c]: Please have another look.

**Comment by jesperpedersen** [comment]
I don't see the `GETTING_STARTED.md` and `pgmoneta.spec` changes...

**Comment by solokyo** [comment]
> I don't see the `GETTING_STARTED.md` and `pgmoneta.spec` changes...

Added the encryption tutorial link to `GETTING_STARTED.md` and `.spec` rpmbuild file.

**Comment by jesperpedersen** [comment]
You need to add `encrypt` and `decrypt` to the help output from `pgmoneta-cli -?`, and then include - as well - in `GETTING.md`. You forgot the new tutorial in the `%files` section of `pgmoneta.spec`.

Plus

```
2022-09-14 11:52:39 WARN  management.c:1460 pgmoneta_management_read_int32: read: 3 Invalid argument
```

```
==583779==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 41 byte(s) in 1 object(s) allocated from:
    #0 0x55e49d4430c0 in __interceptor_realloc.part.0 asan_malloc_linux.cpp.o
    #1 0x7ff6ca8d56ed in pgmoneta_append /home/jews/PostgreSQL/pgmoneta/official/src/libpgmoneta/utils.c:800:15
    #2 0x7ff6ca87b2eb in pgmoneta_encrypt_file /home/jews/PostgreSQL/pgmoneta/official/src/libpgmoneta/aes.c:192:12
    #3 0x55e49d48fcfc in accept_mgt_cb /home/jews/PostgreSQL/pgmoneta/official/src/main.c:1230:10
    #4 0x7ff6caf715d2 in ev_invoke_pending /home/jews/PostgreSQL/pgagroal/libev/ev.c:3770:11

SUMMARY: AddressSanitizer: 41 byte(s) leaked in 1 allocation(s).
```

**Comment by solokyo** [comment]
I'll squash the commit once you have done the review.

**Comment by jesperpedersen** [comment]
Yeah, this looks good.

However, if you do

```
pgmoneta-cli decrypt /path/to/file_that_does_not_exists
```

or

```
pgmoneta-cli encrypt /path/to/file_that_does_not_exists
```

you will get 

```
WARN  management.c:1460 pgmoneta_management_read_int32: read: 3 Invalid argument
```

Please, fix this squash and rebase.

File: src/cli.c [hunk 2]
```diff
@@ -1000,3 +1030,21 @@ decrypt(SSL* ssl, int socket, char* path)
...
+
+static int
+encrypt(SSL* ssl, int socket, char* path)
+{
+   int ret;
+   int number_of_returns = 0;
+   int code = 0;
+
+   ret = pgmoneta_management_encrypt(ssl, socket, path);
+   pgmoneta_management_read_int32(ssl, socket, &number_of_returns);
```
Comment (solokyo) [inline src/cli.c:1044]: I mimicked this piece of code based on decrypt. The `number_of_returns` is 1 on error, but why `number_of_returns` here? Both `decrypt` and `encrypt` should have only one int return value to indicate the exec result. 
The return value of `pgmoneta_decrypt_archive` indicates that we correctly sent the command to the server. But what about the actual exec return value? Which one should be the `$?`

Comment (jesperpedersen) [reply src/cli.c:1044]: Yeah, that is an oversight in that commit. You can do

```
   int ret;

   if (pgmoneta_management_decrypt(ssl, socket, path))
   {
       return 1;
   }
   pgmoneta_management_read_int32(ssl, socket, &ret);
   return ret;
```

So, `0` upon successful decryption, and `1` upon failure

**Comment by jesperpedersen** [comment]
Merged. (I added the shell completion changes)

Thanks for your contribution !

**Pull request closed without merging**